DigitalExcellence/dex-backend
# Feature/225 swagger documentation

## Description

Improved swagger documentation by adding:

- Response codes for each endpoint
- Return model for each endpoint
- Description for each controller
- Description for the general swagger page

## Type of change


## Checklist


## Steps to Test or Reproduce

Outline the steps to test or reproduce the PR here.
These steps will be used during release testing.

1. When you navigate to the swagger page, you should see below Dex API (below '/swagger/v1/swagger.json') a general description for the swagger page for Dex
2. After each controller name, there should be a small description visible for each controller individually
3. In a response tab, you should be able to see under 'Reponses' all the difference response codes with a short description, example value, schema, and media type.
4. All response media types should be set to application/json
5. All request media types should be set to application/json as the default value
5. When placing `[Consumes("")]' attribute for an endpoint, this endpoint won't be set to json automatically because you set your own custom configuration for the endpoint

## Link to issue

Closes: #225
